pkg/parser/ast/ddl.go

```diff
@@ -921,7 +919,6 @@ func (n *Constraint) Restore(ctx *format.RestoreCtx) error {
 	}
 
 	if n.Option != nil {
-		ctx.WritePlain(" ")
 		if err := n.Option.Restore(ctx); err != nil {
```

Comment:
The convention is that "Restore()" always generates a string with whitespace trimming :)

Reply:
I can revert this change if you like, I did it so that the full restore would not add additional spaces in case there were a default IndexOption, like LOCAL.

Reply:
I think we can add `IndexOption.isEmpty()` to help determine whether it is a default value:

```
if n.Option != nil && !n.Option.isEmpty() {
		ctx.WritePlain(" ")
		if err := n.Option.Restore(ctx); err != nil {
```

Then it is consist with other AST nodes, and the change in `runNodeRestoreTestWithFlags` is not necessary.

Reply:
Thanks for the suggestion, done :)